app/models/case/sar/offender.rb

```diff
@@ -454,9 +454,7 @@ def user_dealing_with_vetting
   # Overwrites base method to allow case number to remove "R" when
   # transitioning from 'rejected' to 'valid' offender SAR
   def prevent_number_change
-    return unless current_state != "rejected" && number_changed?
-
-    raise StandardError, "number is immutable"
+    raise StandardError, "number is immutable" if !case_originally_rejected && number_changed?
```

Comment:
Used the case_originally_rejected flag as save is also called from machine.rb write_transition method. The second time prevent_number_change is called the current_state is not rejected which causes the error to be raised which is not wanted.

Reply:
The issue is that there is nothing to stop a previously rejected case having its number changed.

So when `write_transition` is called, `number_changed?` returns true?